python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -582,8 +582,6 @@ def reserve_and_allocate_spread(
                 utilization_scorer=utilization_scorer,
                 strict_spread=True,
             )
-            _inplace_add(node_type_counts, to_launch)
-            _inplace_add(to_add, to_launch)
```

Comment:
Again, we can't override the `node_type_counts` and `to_add` without making sure the strict spread bundle can be scheduled.